### DISCUSSION TITLE
Forking for local testing

### DISCUSSION THREAD

**Discussion by Gilgamesh-818** [zkSync-Community-Hub/zksync-developers#60] [General] [answered]
Is there any way to fork the current state of the zksync network for local node testing? I know hardhat supports it for eth l1, but I can't find anything for era.

**Comment by MexicanAce** [comment, marked as answer]
There's a new local test node that's in early Alpha stages: https://github.com/matter-labs/era-test-node

As it's still being released, there are limitations, so please read the README for the latest information.

**Reply by Gilgamesh-818** [reply]
Thank you!

**Comment by martinerinv** [comment]
Currently, there isn't direct support for forking the zkSync network for local node testing like there is for Ethereum L1 with Hardhat. However, you can manually set up a local development environment using zkSync's testing tools and a local zkSync node. This involves deploying contracts to a local zkSync instance and using tools like Hardhat for testing. While not as streamlined as forking, this approach allows for local development and testing of zkSync applications.

**Comment by tysaboswe** [comment]
Currently, there isn't a direct way to fork the current state of the zkSync network for local node testing similar to what is available for Ethereum L1 using Hardhat. However, zkSync's team is working on providing support for this feature in the near future. Stay tuned for updates!

**Comment by sinanceviccarter** [comment]
Currently, there is no direct support for forking the zkSync network for local node testing similar to what is available for Ethereum L1 with Hardhat. However, you can set up a local development environment using zkSync's testnet and leverage zkSync's development tools and libraries for testing and development purposes.

**Comment by lashawnheisler** [comment]
Currently, there is no direct built-in support for forking the zkSync network for local node testing similar to what Hardhat offers for Ethereum L1. However, developers can set up a local zkSync environment using zkSync's development tools and libraries, such as zksync-rs and zksync-js, to simulate various scenarios and test their applications locally.

**Comment by estacilo** [comment]
Yes, it is possible to fork the current state of the zkSync network for local node testing using the zksync-local package. This package provides a local development environment for zkSync, allowing developers to test their smart contracts and applications in a local environment that mimics the mainnet. However, it is important to note that this package is still in beta and should be used with caution.
